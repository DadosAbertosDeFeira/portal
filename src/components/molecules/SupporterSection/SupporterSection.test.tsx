import { render } from "@testing-library/react";
import type { ImageProps } from "atoms/Image";

import type { Supporter } from "@/models/SupportersModel";

import type { SupporterSectionProps } from ".";
import { SupporterSection } from ".";

jest.mock("next/image", () => ({ src, ...props }: ImageProps) => (
  <img src={src as string} {...props} />
));

describe("Supporters", () => {
  const supportersMock: Supporter = {
    id: "shuttleworth",
    img: "https://supporter.test/image.png",
    link: "https://shuttleworthfoundation.org/",
    name: "Shuttleworth Foundation",
  };

  const makeSut = ({
    title = "Apoio",
    supporters = [supportersMock],
    ...props
  }: Partial<SupporterSectionProps>) => {
    return render(
      <SupporterSection title={title} supporters={supporters} {...props} />
    );
  };

  it("renders correctly", () => {
    const { asFragment } = makeSut({});

    expect(asFragment).toMatchSnapshot();
  });
});
