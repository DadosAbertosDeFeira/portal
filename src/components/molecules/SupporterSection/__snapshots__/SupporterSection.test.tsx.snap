// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supporters renders correctly 1`] = `[Function]`;
